lms/djangoapps/learner_home/serializers.py

```diff
@@ -239,23 +255,43 @@ def get_certPreviewUrl(self, enrollment):
 class AvailableEntitlementSessionSerializer(serializers.Serializer):
     """An available entitlement session"""
 
-    startDate = serializers.DateTimeField()
-    endDate = serializers.DateTimeField()
-    courseId = serializers.CharField()
+    startDate = serializers.DateTimeField(source='start')
+    endDate = serializers.DateTimeField(source='end')
+    courseId = serializers.CharField(source='key')
 
 
 class EntitlementSerializer(serializers.Serializer):
     """Entitlements info"""
+    requires_context = True
 
-    availableSessions = serializers.ListField(
-        child=AvailableEntitlementSessionSerializer(), allow_empty=True
-    )
-    isRefundable = serializers.BooleanField()
-    isFulfilled = serializers.BooleanField()
-    canViewCourse = serializers.BooleanField()
-    changeDeadline = serializers.DateTimeField()
-    isExpired = serializers.BooleanField()
-    expirationDate = serializers.DateTimeField()
+    availableSessions = serializers.SerializerMethodField()
+    uuid = serializers.UUIDField()
+    isRefundable = serializers.BooleanField(source='is_entitlement_refundable')
+    isFulfilled = serializers.SerializerMethodField()
+    changeDeadline = serializers.SerializerMethodField()
+    isExpired = serializers.SerializerMethodField()
+    expirationDate = serializers.SerializerMethodField()
+
+    # For some reason BooleanField won't cast None values to False so I'm hacking this
```

Comment:
Would a  [NullBooleanField]( https://www.django-rest-framework.org/api-guide/fields/#nullbooleanfield ) solve the issue?

Reply:
No, NullBooleanField is the field that's supposed to allow null. I have no clue why BooleanField is allowing null